rusk-wallet: Implement moonlight history

Daksh14 opened the pull request:
Closes #2271

herr-seppia reviewed (changes requested):
Would be possible to have a tx history containing both Phoenix and moonlight transactions?
Why did you split them?

Additionally (but not blockers):
- try to store the history in rocksdb to improve performance and unload archival computation (we already store Notes, you only miss the derived TransactionHistory objects)
- add a warning if the user is not connected to an archival

Daksh14 commented:
> Would be possible to have a tx history containing both Phoenix and moonlight transactions?
> Why did you split them?
> 
> Additionally (but not blockers):
> - try to store the history in rocksdb to improve performance and unload archival computation (we already store Notes, you only miss the derived TransactionHistory objects)
> - add a warning if the user is not connected to an archival

Mixing the moonlight and phoenix tx history would be confusing no? I can add another column that displays transaction type but that would add more info and make it harder to read

I split the history because all other methods are also split on public/private it makes sense.

Can you create issue for those two points you mentioned?

Neotamandua commented:
> Mixing the moonlight and phoenix tx history would be confusing no? I can add another column that displays transaction type but that would add more info and make it harder to read

In the long run, I think it is necessary to add such a column anyway. Because we want to differentiate between different types of transactions (transfer out, transfer in, conversion, deposit, withdrawal, etc.).

herr-seppia commented:
> Because we want to differentiate between different types of transactions (transfer out, transfer in, conversion, deposit, withdrawal, etc.).

These infos are already available in the transaction history in the "method" column IIRC

Daksh14 commented:
@herr-seppia lets keep the moonlight and phoenix history seperate for now, we dont have to sync if we're just fetching moonlight history

Neotamandua commented on rusk-wallet/src/bin/command.rs:
```diff
@@ -514,13 +512,23 @@ impl Command {
+                {
+                    phoenix_history.append(&mut moonlight_history);
+                } else {
+                    tracing::error!("Cannot fetch archival history");
```
```suggestion
                    tracing::error!("Cannot fetch archive history");
```

Neotamandua commented on rusk-wallet/src/bin/command.rs:
```diff
@@ -514,13 +512,23 @@ impl Command {
+
                 wallet.sync().await?;
                 let notes = wallet.get_all_notes(profile_idx).await?;
+                let public_key = wallet.public_address(profile_idx)?;
```
```suggestion
                let address = wallet.public_address(profile_idx)?;
```

Neotamandua commented on rusk-wallet/src/error.rs:
```diff
@@ -162,6 +162,9 @@ pub enum Error {
     #[error("Inquire error: {0}")]
     InquireError(String),
+    /// Error while querying archival node
+    #[error("Archival node query error: {0}")]
+    ArchivalJsonError(String),
```
```suggestion
    #[error("Archive node query error: {0}")]
    ArchiveJsonError(String),
```
